backend/sqlite/sqlite_test.go

```diff
@@ -36,3 +41,74 @@ func Test_EndToEndSqliteBackend(t *testing.T) {
 		require.NoError(t, b.Close())
 	})
 }
+
+func Test_SqliteBackend_WorkerName(t *testing.T) {
+	t.Run("DefaultWorkerName", func(t *testing.T) {
+		backend := NewInMemoryBackend()
+		defer backend.Close()
+
+		// The default worker name should be in the format "worker-<uuid>"
+		require.Contains(t, backend.workerName, "worker-")
+		require.Len(t, backend.workerName, 43) // "worker-" (7) + UUID (36)
+	})
+
+	t.Run("CustomWorkerName", func(t *testing.T) {
+		customWorkerName := "test-worker-123"
+		backend := NewInMemoryBackend(WithBackendOptions(backend.WithWorkerName(customWorkerName)))
+		defer backend.Close()
+
+		require.Equal(t, customWorkerName, backend.workerName)
+	})
+
+	t.Run("EmptyWorkerNameUsesDefault", func(t *testing.T) {
+		backend := NewInMemoryBackend(WithBackendOptions(backend.WithWorkerName("")))
+		defer backend.Close()
+
+		// Empty worker name should fall back to UUID generation
+		require.Contains(t, backend.workerName, "worker-")
+		require.Len(t, backend.workerName, 43) // "worker-" (7) + UUID (36)
+	})
+
+	t.Run("CustomWorkerNameIsUsedInDatabase", func(t *testing.T) {
+		customWorkerName := "integration-test-worker"
+		backend := NewInMemoryBackend(WithBackendOptions(backend.WithWorkerName(customWorkerName)))
+		defer backend.Close()
+
+		// Verify the worker name is stored correctly
+		require.Equal(t, customWorkerName, backend.workerName)
+
+		// Create a workflow instance and task to ensure the worker name is actually used
+		ctx := context.Background()
+		instance := core.NewWorkflowInstance("test-instance", "test-execution")
+		
+		event := history.NewPendingEvent(
+			time.Now(),
+			history.EventType_WorkflowExecutionStarted,
+			&history.ExecutionStartedAttributes{
+				Queue: "test-queue",
+				Metadata: &workflow.Metadata{},
+			},
+		)
+
+		// Create workflow instance
+		err := backend.CreateWorkflowInstance(ctx, instance, event)
+		require.NoError(t, err)
+
+		// Get a workflow task (this should lock it with our custom worker name)
+		task, err := backend.GetWorkflowTask(ctx, []workflow.Queue{"test-queue"})
+		require.NoError(t, err)
+		require.NotNil(t, task)
+
+		// Query the database to verify our custom worker name is used
+		rows, err := backend.db.Query("SELECT worker FROM instances WHERE id = ? AND execution_id = ?", 
+			instance.InstanceID, instance.ExecutionID)
+		require.NoError(t, err)
+		defer rows.Close()
+
+		var workerNameFromDB string
+		require.True(t, rows.Next())
+		err = rows.Scan(&workerNameFromDB)
+		require.NoError(t, err)
+		require.Equal(t, customWorkerName, workerNameFromDB)
```

Comment:
Just checking that the worker name was passed into `backend.workerName` is enough